src/experimental/tools/download2go/index.ts

```diff
@@ -0,0 +1,208 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AsyncSubject, ReplaySubject } from "rxjs";
+
+import { downloader } from "./apis/publicApi/download";
+import { constructOfflineManifest } from "./apis/publicApi/offlineDownload";
+import { setUpDb } from "./apis/transactionDB/dbSetUp";
+import {
+  checkForPauseAMovie,
+  checkForResumeAPausedMovie,
+  checkForSettingsAddMovie,
+  IndexDBError,
+} from "./utils";
+
+import EventEmitter from "../../../utils/event_emitter";
+import { IActiveSubs, IPauseSubject } from "./apis/dash/types";
+import {
+  IAddMovie,
+  IEventsEmitter,
+  IOptionsStarter,
+  IProgressBarBuilderAbstract,
+  IPublicAPI,
+  IStoredManifest,
+} from "./types";
+
+export default async function D2G({
+  nameDB = "d2g",
+  storeManifestEvery,
+}: IOptionsStarter = {}): Promise<IPublicAPI> {
+  const emitter = new EventEmitter<IEventsEmitter>();
+  const db = await setUpDb(nameDB);
+  if (!db) {
+    throw new IndexDBError("A problem occured during the set up of IndexDB");
+  }
+  const activeSubsDownloader: IActiveSubs = {};
+  const activePauseSubject: IPauseSubject = {};
+
+  return {
+    emitter,
+    async download(settings: IAddMovie): Promise<any> {
+      try {
+        await checkForSettingsAddMovie(settings, db, activeSubsDownloader);
+        const size = 0;
+        const progressBarBuilder = {
+          progress: 0,
+          overall: 0,
+          downloadedID: [],
+        };
+        const progress$ = new ReplaySubject<IProgressBarBuilderAbstract>(1);
+        const pause$ = new AsyncSubject<void>();
+        const { contentID } = settings.dbSettings;
+        activePauseSubject[contentID] = pause$;
+        const subcriptionDownloader = downloader(
+          { ...settings, type: "start" },
```

Comment:
Could you find a way to give meaning to the `downloader` input object ? As a reader, it is difficult to understand what really are input arguments.

Reply:
I'm use to avoid creating a new variable if we don't use it more than 1 time, for performance reason, do I'm right ? or for this kind of this things we can just use a variable because it's more readable and the cost of creating the new variable is so minimal ?